contrib/IECoreUSD/include/IECoreUSD/USDScene.h

```diff
@@ -0,0 +1,105 @@
+
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2017, Image Engine Design Inc. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+//     * Neither the name of Image Engine Design nor the names of any
+//       other contributors to this software may be used to endorse or
+//       promote products derived from this software without specific prior
+//       written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#ifndef IECOREUSD_USDSCENE_H
+#define IECOREUSD_USDSCENE_H
+
+
+#include "pxr/usd/usd/prim.h"
+
+#include "IECore/SceneInterface.h"
+#include "IECoreUSD/TypeIds.h"
+
+namespace IECoreUSD
+{
+
+IE_CORE_FORWARDDECLARE( USDScene );
+
+IE_CORE_DECLAREPTR( USDScene )
+
+class USDScene : public IECore::SceneInterface
```

Comment:
I started by deriving from SampledSceneInterface but took a shortcut and just implemented the SceneInterface reading functions - which would you advise?

Reply:
There are two reasons to use SampledSceneInterface :

- The base class implements interpolation for you automatically if you just implement the sample reading methods (this is how the AlembicScene and SceneCache work). My understanding was that USD had interpolation built in under the hood though, so this might not be useful here.
- I think some spots in Cortex optimise for the SampledSceneInterface case by querying the number of samples to see if there's only one. If we really care about this it might be better to have an `isAnimated()` method in the base class instead anyway - that would be cleaner than the current approach. Gaffer doesn't care at all either way - it uses the hash to optimise the static/animated cases.

As long as my assumption about USD already doing the interpolation is correct, I think deriving straight from SceneInterface is totally fine.

Reply:
Great. I'll keep it as is for now.